### ISSUE TITLE
How to grant run rights to a group of users

### ISSUE THREAD

**Issue by TimotheeMathieu** [sylabs/singularity#974]
Is there a way to grant the right to a group of users to use the same instance of a container ?

Ideally, i would want to be able to launch an instance and then, any user of a given group should be able to use 
`singularity run instance://myinstance`.

**Comment by dtrudg** [comment]
Hi @TimotheeMathieu - I'm afraid this is not possible in the current design.

Singularity's security model relies on container processes running as the user who started the container (instance) and performs any subsequent execution within the container. Access to host files etc. is all dependent on this and host account permissions.

To allow another user to run a process in the container would be analogous to allowing 'sudo' to the account of the user who launched the container.

While this isn't something we can allow easily, it might be useful to understanda a bit more about the circumstances of the workflow that you have? Also, the Slack channel and Google group may be a good venue for more discussion, as they are watched by more of the user community than a single issue. Links to those at: https://sylabs.io/singularity/

**Comment by TimotheeMathieu** [comment]
My goal was to have a container running as an instance on a server. And I wanted to allow all the people from a group to run a command in this container.

In the end, I solved my problem by creating a dummy account, let us call him `dummy`, and adding the following line to sudoers:

```
%mygroup ALL=(root) NOPASSWD: /sbin/su dummy
%mygroup ALL=(root) NOPASSWD: /sbin/su dummy -c "singularity run instance://myinstance"
```

I think this is mostly safe (I hope I did not introduce too many security issues with this). Then, any of the users in the group mygroup is able to log in as the dummy user and use the container.
